docs/Testing.md

```diff
@@ -1,13 +1,134 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+fastify.js
+
+```js 
+const fastify = require('fastify')()
+fastify.get('/', async (request, reply) => {
+  return { hello: 'world' }
+})
+module.exports = fastify
```

Comment:
This is not correct. In order to have a clean separation of concerns, you need to have a function that "builds" the app instance. This is likely the biggest step for a maintainable application and most tutorials etc get it wrong.

Reply:
Thank you for the clarification! I am new to testing but I see it's merits and want to learn as much as possible.

So would it make more sense to wrap this in a function?

```js
const Fastify = require('fastify')

function buildFastify () {
  const fastify = Fastify()
  fastify.get('/', async (request, reply) => {
    return { hello: 'world' }
  })

return fastify
}

module.exports = fastify
```

Reply:
Updated the examples to use a function that builds the app instance